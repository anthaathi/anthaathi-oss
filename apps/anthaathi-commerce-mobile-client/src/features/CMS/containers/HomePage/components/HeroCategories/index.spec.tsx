import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import HeroCategories from './index';

describe('HeroCategories', () => {
  it('should render the item', function () {
    const temp = render(
      <IntlProvider locale="en-US" messages={locale}>
        <ThemeProvider>
          <HeroCategories
            title="Test"
            items={[
              {
                title: 'test',
                title_ar: 'test',
                key: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </ThemeProvider>
      </IntlProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('heroCategory')).toBeTruthy();
  });

  it('should fire event when click on the categories', function () {
    const test = jest.fn();

    const temp = render(
      <IntlProvider locale="en-US" messages={locale}>
        <ThemeProvider>
          <HeroCategories
            onPress={test}
            title="Test"
            items={[
              {
                title: 'test',
                key: 'test',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
              {
                title: 'test',
                key: 'test2',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </ThemeProvider>
      </IntlProvider>,
    );

    expect(temp).toMatchSnapshot();
    fireEvent.press(temp.queryByTestId('heroItemtest')!);
    expect(test).toBeCalledWith('test');
  });

  it('renders expected number of items', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroCategories
            title="Test"
            items={[
              {
                title: 'test',
                key: 'test',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
              {
                title: 'test',
                key: 'test2',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    const getTestId = temp.queryByTestId('heroCategoriesList')!;
    expect(getTestId.props.data.length).toBe(1);
    expect(getTestId.props.data[0][0]).toEqual({
      title: 'test',
      key: 'test',
      title_ar: 'test',
      image:
        'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
    });
  });

  it('should have HeroCategories title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroCategories
            title="Test"
            items={[
              {
                title: 'test',
                key: 'test',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
              {
                title: 'test',
                key: 'test2',
                title_ar: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    expect(temp.queryByText('Test')).toBeTruthy();
  });
});
