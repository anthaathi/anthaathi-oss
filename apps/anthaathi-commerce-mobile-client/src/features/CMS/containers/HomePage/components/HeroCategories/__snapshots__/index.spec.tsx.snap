// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCategories should fire event when click on the categories 1`] = `
<View
  testID="heroCategory"
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 9,
          "marginHorizontal": 6,
          "marginVertical": 6,
        },
      ]
    }
  >
    Test
  </Text>
  <View>
    <RCTScrollView
      data={
        [
          [
            {
              "image": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
              "key": "test",
              "title": "test",
              "title_ar": "test",
            },
            {
              "image": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
              "key": "test2",
              "title": "test",
              "title_ar": "test",
            },
          ],
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={4}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="heroCategoriesList"
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "column",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  false,
                  {
                    "borderRadius": 12,
                    "marginRight": 12,
                    "padding": 4,
                  },
                ]
              }
              testID="heroItemtest"
            >
              <View>
                <Image
                  source={
                    {
                      "height": 97,
                      "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
                      "width": 97,
                    }
                  }
                  style={
                    {
                      "borderRadius": 48.5,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "fontWeight": "600",
                        "marginBottom": 12,
                        "marginTop": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                  variant="labelMedium"
                >
                  test
                </Text>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  false,
                  {
                    "borderRadius": 12,
                    "marginRight": 12,
                    "padding": 4,
                  },
                ]
              }
              testID="heroItemtest2"
            >
              <View>
                <Image
                  source={
                    {
                      "height": 97,
                      "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
                      "width": 97,
                    }
                  }
                  style={
                    {
                      "borderRadius": 48.5,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "fontWeight": "600",
                        "marginBottom": 12,
                        "marginTop": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                  variant="labelMedium"
                >
                  test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`HeroCategories should render the item 1`] = `
<View
  testID="heroCategory"
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 9,
          "marginHorizontal": 6,
          "marginVertical": 6,
        },
      ]
    }
  >
    Test
  </Text>
  <View>
    <RCTScrollView
      data={
        [
          [
            {
              "image": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
              "key": "test",
              "title": "test",
              "title_ar": "test",
            },
          ],
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={4}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="heroCategoriesList"
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "column",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  false,
                  {
                    "borderRadius": 12,
                    "marginRight": 12,
                    "padding": 4,
                  },
                ]
              }
              testID="heroItemtest"
            >
              <View>
                <Image
                  source={
                    {
                      "height": 97,
                      "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
                      "width": 97,
                    }
                  }
                  style={
                    {
                      "borderRadius": 48.5,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "fontWeight": "600",
                        "marginBottom": 12,
                        "marginTop": 6,
                        "textAlign": "center",
                      },
                    ]
                  }
                  variant="labelMedium"
                >
                  test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
